src/compiler/transformers/es2015.ts

```diff
@@ -3173,7 +3173,7 @@ namespace ts {
         function visitCatchClause(node: CatchClause): CatchClause {
             const ancestorFacts = enterSubtree(HierarchyFacts.BlockScopeExcludes, HierarchyFacts.BlockScopeIncludes);
             let updated: CatchClause;
-            if (isBindingPattern(node.variableDeclaration.name)) {
+            if (node.variableDeclaration && isBindingPattern(node.variableDeclaration.name)) {

```

Comment:
By the time we get to the es2015 transform we **must** already have a variableDeclaration. Filling in the missing binding should be handled in the esnext transform. At best we should have a debug assertion.